docs/02-app/01-building-your-application/08-deploying/01-production-checklist.mdx

```diff
@@ -0,0 +1,138 @@
+---
+title: Going to Production
+description: Before taking your Next.js application to production, here are some recommendations to ensure the best user experience.
+---
+
+Before taking your Next.js application to production, here are some recommendations to ensure the best user experience.
+
+## Overview
+
+- Use [caching](/docs/app/building-your-application/caching) wherever possible.
+- Ensure your database and backend are deployed in the same region.
+- Focus on leveraging [Server Components](https://nextjs.org/docs/app/building-your-application/rendering/server-components) for efficient rendering. Server Components enable rendering on the server, which can significantly reduce the size of your client-side JavaScript bundle. Remember, Server Components can contain Client Components, offering flexibility in how you compose your application's UI. For an in-depth understanding of security best practices with Server Components, see our [blog post on security in Next.js Server Components and Actions](https://nextjs.org/blog/security-nextjs-server-components-actions).
+- Prevent sensitive data from being exposed to the client with [tainting](/docs/app/building-your-application/data-fetching/patterns#preventing-sensitive-data-from-being-exposed-to-the-client)
+- Ensure errors in nested route segments are handled gracefully with the [error.js file](https://nextjs.org/docs/app/building-your-application/routing/error-handling)
+- Customize your 404 (Not Found) responses using the [`notFound()`](/docs/app/api-reference/functions/not-found) function and [`not-found.js` file](https://nextjs.org/docs/app/api-reference/file-conventions/not-found)
+- Ensure you are [measuring performance](/docs/app/building-your-application/optimizing/analytics).
```

Comment:
This point could come after the following one, e.g.
- use lighthouse to test your product's performance
- on production, collect real user metrics (RUM) with analytics